src/keyword.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+/// Just like Clojure's Keyword. We'll need this as part of EDN parsing,
```

Comment:
This might end up being a cyclic dependency if it's needed for both EDN parsing and core but defined in core.  The implementation probably makes sense in a new edn-parser crate and it can be imported into core from that crate if it's needed here.

If you want to start building out the edn-parser in core and break it into it's own crate later that's fine, just keeping in mind that it might become more difficult the longer we wait.